semaphore.cpp

```diff
@@ -1,23 +1,24 @@
 #include "semaphore.h"
 
-
 Semaphore::Semaphore(int initialValue)
     :mSemaphoreValue(initialValue)
 {
-
 }
 
-void Semaphore::Increment() //Signal - Unlock
+void Semaphore::Signal() //Signal - Unlock
 {
     mSemaphoreValue++;
-    mMutexObject.unlock();
+    mConditionVariable.notify_all();
 }
 
-void Semaphore::Decrement() //Wait - Lock
+void Semaphore::Wait() //Wait - Lock
 {
     mSemaphoreValue--;
-    if (mSemaphoreValue < 0)
+    if (mSemaphoreValue >= 0)
     {
```

Comment:
After reading cppreference for std::atomic, I declared this as 'std::atomic<int> mSemaphoreValue' in order to avoid race conditions. Do you refer to something else ?

Reply:
So a lock is still necessary, decrement value at the end, and replace if statement with 'while (mSemaphoreValue <= 0)... lock'